File: examples/snippets/modules/ROOT/examples/build-flink-streamlets-scala/step1/src/main/scala/com/example/ReportPrinter.scala
package com.example

import cloudflow.streamlets.avro._
import cloudflow.streamlets.StreamletShape
import cloudflow.flink._

// TODO rename to ReportPrinter
class ReportPrinterStep1 extends FlinkStreamlet {
  // 1. Create inlets and outlets
  @transient val in = AvroInlet[Report]("report-in")

  // 2. TODO Define the shape of the streamlet
  val shape = StreamletShape.empty
  // 3. TODO Override createLogic to provide StreamletLogic
  def createLogic(): FlinkStreamletLogic = new FlinkStreamletLogic() {
    def buildExecutionGraph = ()
  }
}

File: examples/snippets/modules/ROOT/examples/build-flink-streamlets-java/step1/src/main/java/com/example/ReportPrinter.java
package com.example;

import cloudflow.streamlets.StreamletShape;
import cloudflow.streamlets.avro.*;
import cloudflow.flink.*;

public class ReportPrinter extends FlinkStreamlet {
  // 1. Create inlets and outlets
  transient AvroInlet<Report> in = AvroInlet.<Report>create("report-in", Report.class);

  // 2. TODO Define the shape of the streamlet
  public StreamletShape shape() { return StreamletShape.empty(); }
  // 3. TODO Override createLogic to provide StreamletLogic
  public FlinkStreamletLogic createLogic() { 
    return new FlinkStreamletLogic(getContext()) {
      public void buildExecutionGraph() {
      }
    };
  }
}